certification/internal/shell/has_minimal_vulns.go

```diff
@@ -1,15 +1,96 @@
 package shell
 
 import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
-	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	ovalFilename = "rhel-8.oval.xml.bz2"
+	ovalUrl      = "https://www.redhat.com/security/data/oval/v2/RHEL8/"
+	reportFile   = "vuln.html"
 )
 
 type HasMinimalVulnerabilitiesCheck struct{}
 
 func (p *HasMinimalVulnerabilitiesCheck) Validate(image string) (bool, error) {
-	return false, errors.ErrFeatureNotImplemented
+
+	ovalFileUrl := fmt.Sprintf("%s%s", ovalUrl, ovalFilename)
+	dir, err := ioutil.TempDir(".", "oval-")
+
+	if err != nil {
+		log.Error("Unable to create temp dir", err)
+		return false, err
+	}
+	log.Debugf("Oval file dir: %s", dir)
+	defer os.RemoveAll(dir)
+
+	ovalFilePath := filepath.Join(dir, ovalFilename)
+	log.Debugf("Oval file path: %s", ovalFilePath)
+
+	err = file.DownloadFile(ovalFilePath, ovalFileUrl)
+	if err != nil {
+		log.Error("Unable to download Oval file", err)
+		return false, err
+	}
+	// get the file name
+	r := regexp.MustCompile(`(?P<filename>.*).bz2`)
+	ovalFilePathDecompressed := filepath.Join(dir, r.FindStringSubmatch(ovalFilename)[1])
+
+	err = file.Unzip(ovalFilePath, ovalFilePathDecompressed)
+	if err != nil {
+		log.Error("Unable to unzip Oval file: ", err)
+		return false, err
+	}
+
+	numOfVulns, err := p.numberOfVulnerabilities(image, ovalFilePathDecompressed)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
+	}
+
+	return true, nil
 }
+
+func (p *HasMinimalVulnerabilitiesCheck) numberOfVulnerabilities(image string, ovalFilePathDecompressed string) (int, error) {
+
+	cmd := exec.Command("oscap-podman", image, "oval", "eval", "--report", reportFile, ovalFilePathDecompressed)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+
+	if err != nil {
+		log.Error("unable to execute oscap-podman on the image: ", cmd.Stderr, reportFile)
+		return 0, err
+	}
+
+	path, err := os.Getwd()
+	if err != nil {
+		log.Error("unable to get the current directory: ", err)
+	}
```

Comment:
Should we be returning this error here, or is it safe to continue?

Reply:
It is safe to continue in my opinion. The vulnerability report will be written to the current directory, but the check does not use the report to find out the number of vulnerabilities. The report provides a human readable HTML report to see the scanning result.